### PR TITLE
add support for kafka streams new processing api

### LINKED ISSUE #1365: KafkaStreamsTracing update to align with new Kafka Streams API
## Feature
The current "Transform" and "Process" methods on the Kafka Streams KStream class are now deprecated in favour of simpler "Process" and "ProcessValues" methods. Is there any plan to update the existing KafkaStreamsTracing class to emulate this new approach?

### PR DESCRIPTION
**Pull request by dermotmburke**
Fix #1365

### CODE REVIEW

File: instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java
Comment (jeqo) [inline instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java:525]: So here it would be:
```suggestion
    return new TracingV2FixedKeyProcessorSupplier<>(this, spanName, processorSupplier);
```

**Review by jeqo — COMMENTED** [review]
Thanks @dermotmburke! This looks great!
I took an initial look and added some comments about naming, though I'm open to hear other opinions.

File: instrumentation/kafka-streams/src/test/java/brave/kafka/streams/KafkaStreamsTest.java
Comment (jeqo) [inline instrumentation/kafka-streams/src/test/java/brave/kafka/streams/KafkaStreamsTest.java:76]: nit: Don't forget to align the variable names to the renamed class.

**Review by jeqo — CHANGES_REQUESTED** [review]
Thanks @dermotmburke. I guess you missed changing the file names :) 

Once build is fixed and names aligned, we could wrap up this PR by updating the instrumentation/kafka-streams/README.md and RATIONALE.md. Let me know how you feel about adding this and if you need any help. Thanks again!

**Comment by dermotmburke** [comment]
Apologies @jeqo I should have checked the PR changes a bit closer :) 

I have updated the README.md and RATIONALE.md files as suggested. Wasn't entirely sure what changes to make so let me know if they make sense.

Thanks

File: instrumentation/kafka-streams/RATIONALE.md
Comment (jeqo) [inline instrumentation/kafka-streams/RATIONALE.md:11]:
```suggestion
the Producers at the middle (e.g. `builder.through(topic)` or `builder.repartition()`) or at the end of a Stream topology
```

File: instrumentation/kafka-streams/RATIONALE.md
Comment (jeqo) [inline instrumentation/kafka-streams/RATIONALE.md:32]:
```suggestion
code involved to disable tracing may mean more code than vice versa!
```

File: instrumentation/kafka-streams/README.md
Comment (jeqo) [inline instrumentation/kafka-streams/README.md:21]:
```suggestion
[KIP-820](https://cwiki.apache.org/confluence/display/KAFKA/KIP-820%3A+Extend+KStream+process+with+new+Processor+API)
introduces new processor APIs to the Kafka Streams DSL.
The following sections show how to instrument applications with the latest and previous APIs.

## Kafka Streams  >= v3.4.0
```

**Review by jeqo — APPROVED** [review]
Thanks @dermotmburke! Left some minor comments to improve docs, but all looks good to me. Once fixed, it should be good to merge unless CI says otherwise :)

File: instrumentation/kafka-streams/RATIONALE.md
Comment (jcchavezs) [reply instrumentation/kafka-streams/RATIONALE.md:32]: I think this was Adrian's signature.

Comment (jeqo) [reply instrumentation/kafka-streams/RATIONALE.md:32]: 😂 then ignore!

**Comment by dermotmburke** [comment]
Sorry - been a bit too busy with family stuff over the Easter period to look at this. All looks good to me. Thanks

**Review by jeqo — APPROVED** [review]
thanks @dermotmburke !

Let's wait a bit if there's additional feedback from the the team, but LGTM!

**Comment by dermotmburke** [comment]
Hi @jeqo - looks like there was an error running the gRPC tests - is this a known issue?

**Comment by jeqo** [comment]
Thanks for noticing! I don't remember to have seeing it before. but it's definitely unrelated. Retrying now, let's see if persists

File: instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingV2Processor.java
Comment (jcchavezs) [inline instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingV2Processor.java:69]: I wonder why this does not happen in the catch and if we should check for noop.

File: instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingV2FixedKeyProcessor.java
Comment (jcchavezs) [inline instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingV2FixedKeyProcessor.java:69]: same comment as below.

File: instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingV2Processor.java
Comment (jeqo) [reply instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingV2Processor.java:69]: Interesting. Maybe we can discuss this in another issue (if it's an issue)?. 
I can see we have follow this pattern on kafka-streams and kafka-clents:
- https://github.com/openzipkin/brave/blob/0e26fd75093d3e42600853ecdb6f95a0157d553e/instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingTransformer.java#L64
- https://github.com/openzipkin/brave/blob/f924f7f1541652c6c547570db81ed9bbaf54aac0/instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingCallback.java#L41

**Review by jcchavezs — APPROVED** [review]
Thanks @dermotmburke!

**Comment by jeqo** [comment]
Thanks @jcchavezs and @dermotmburke for the great work!

**Comment by jeqo** [comment]
Will ship a new release later today 🚀

**Comment by dermotmburke** [comment]
Thanks for all the support @jeqo

**Comment by jcchavezs** [comment]
<3

**Pull request merged**
